File: Python/src/day52/Exercise.py
import math
class Exercise:
    class ClassComplexNumber:
        def __init__(self, real, imaginary):
            self.real = real
            self.imaginary = imaginary

        def __add__(self, other):
            return Exercise.ClassComplexNumber(self.real + other.real, self.imaginary + other.imaginary)

        def __sub__(self, other):
            return Exercise.ClassComplexNumber(self.real - other.real, self.imaginary - other.imaginary)

        def __mul__(self, other):
            real = self.real * other.real - self.imaginary * other.imaginary
            imaginary = self.real * other.real + self.imaginary * other.imaginary
            return Exercise.ClassComplexNumber(real, imaginary)

        def __truediv__(self, other):
            r = (self.real * other.real + self.imaginary * other.imaginary) / (other.real * other.real + other.imaginary * other.imaginary)
            i = (self.imaginary * other.real - self.real * other.imaginary) / (other.real * other.real + other.imaginary * other.imaginary)
            return Exercise.ClassComplexNumber(r, i)

        def __floordiv__(self, other):
            r = (self.real * other.real + self.imaginary * other.imaginary) // (other.real * other.real + other.imaginary * other.imaginary)
            i = (self.imaginary * other.real - self.real * other.imaginary) // (other.real * other.real + other.imaginary * other.imaginary)
            return Exercise.ClassComplexNumber(r, i)

    class Circle:
        def __init__(self, radius):
            self.radius = radius
        def area(self):
            return math.pi * self.radius * self.radius
        def perimeter(self):
            return 2 * math.pi * self.radius

    class Rectangle:
        def __init__(self, length, width):
            self.length = length
            self.width = width
        def area(self):
            return self.length * self.width
        def perimeter(self):
            return 2 * (self.length + self.width)

File: C#/src/Intermediate/day52/Exercise.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Intermediate.day52
{
    public class Exercise
    {
        public class ClassComplexNumber
        {
            float real, imaginary;
            ClassComplexNumber(float x, float y)
            {
                this.real = x;
                this.imaginary = y;
            }
            ClassComplexNumber add(ClassComplexNumber x)
            {
                // Complete the function
                return new ClassComplexNumber(this.real + x.real, this.imaginary + x.imaginary);
            }

            ClassComplexNumber subtract(ClassComplexNumber x)
            {
                // Complete the function
                return new ClassComplexNumber(this.real - x.real, this.imaginary - x.imaginary);
            }

            ClassComplexNumber multiply(ClassComplexNumber x)
            {
                // Complete the function
                float r = x.real * this.real - x.imaginary * this.imaginary;
                float i = this.imaginary * x.real + this.real * x.imaginary;
                return new ClassComplexNumber(r, i);
            }

            ClassComplexNumber divide(ClassComplexNumber x)
            {
                // Complete the function
                float r = (this.real * x.real + this.imaginary * x.imaginary) / (x.real * x.real + x.imaginary * x.imaginary);
                float i = (this.imaginary * x.real - this.real * x.imaginary) / (x.real * x.real + x.imaginary * x.imaginary);
                return new ClassComplexNumber(r, i);
            }
        }
        class Circle
        {
            // Define properties here
            float radius;
            float pi = 3.14f;

            // Define constructor here
            public Circle(float radius)
            {
                this.radius = radius;
            }

            float perimeter()
            {
                // Complete the function
                return 2 * pi * this.radius;
            }

            float area()
            {
                // Complete the function
                return pi * this.radius * this.radius;
            }
        }
        class Rectangle
        {
            // Define properties here
            int x, y;

            // Define constructor here
            public Rectangle(int x, int y)
            {
                this.x = x;
                this.y = y;
            }

            int perimeter()
            {
                // Complete the function
                return 2 * (x + y);
            }

            int area()
            {
                // Complete the function
                return x * y;
            }
        }
    }
}
